doc/modules/array_api.rst

```diff
@@ -141,6 +141,7 @@ Metrics
 -------
 
 - :func:`sklearn.metrics.accuracy_score`
+- :func:`sklearn.metrics.confusion_matrix`
```

Comment:
I think we should have another section where we specify metrics that don't support the array api in internal computations and use numpy instead because the array api does not support the optimizations needed. Therefore using numpy is more appropriate internally and we convert the output back to the desired array.
This should be done because we don't guarantee and speedups for such metrics.

Reply:
I think it's fine to not add the additional document in this PR. But don't you think that where other metrics that are documented use the array api internally and the ones involving confusion matrix or confusion matrix itself don't use the array api, it would be better to clarify such a point? I think we can ask for opinions from others as well.

Reply:
Just following up on this after reviewing https://github.com/scikit-learn/scikit-learn/pull/29251 - do you think it would be useful to document the metrics that do convert to numpy internally, not from a 'justification' perspective, but to document for maintainers as it may be useful at some point later?